### PR TITLE
Dr/oauth2 improvements

### PR DESCRIPTION
**Pull request by remenska**
* improved explanation on OAuth 2.0 roles&flows, and added a sequence diagram

### CODE REVIEW

File: src/source/authentication.rst [hunk 1]
```diff
@@ -235,7 +244,15 @@ After obtaining the authorization code at step 2 the client needs to make a
     Host: admarkt.demo.qa-mp.so
     Content-Type: application/x-www-form-urlencoded
 
-    grant_type=authorization_code&code=AUTH_CODE&client_id=YOUR_CLIENT_ID&client_secret=YOUR_CLIENT_SECRET&redirect_uri=https://yoursite.com/code
+    -d 'grant_type=authorization_code&code=AUTH_CODE&client_id=YOUR_CLIENT_ID&client_secret=YOUR_CLIENT_SECRET&rredirect_uri=https://yoursite.com/code' https://admarkt.marktplaats.nl/accounts/oauth/token
```
Comment (mainnika) [inline src/source/authentication.rst]: this probably not correct if we say about post body, should be like it was before.
I see here a post body request, first headers, then data. it is not a curl cli example.

File: src/source/authentication.rst [hunk 2]
```diff
@@ -125,6 +130,8 @@ Getting an Access Token
 
 The steps for getting an access token are as follows:
 
+If the user is not logge-in with the tenant:
```
Comment (twessling-icas) [inline src/source/authentication.rst]:
```suggestion
If the user is not logged-in with the tenant:
```

File: src/source/authentication.rst [hunk 3]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
   In this context *authorization server* is the server hosting authentication
   and token endpoints.
 
+In our Oauth2 flow we rely on the *tenant* as being the **Authentication Server**, the
```
Comment (twessling-icas) [inline src/source/authentication.rst]: Do we want to call this 'tenant' or 'local market' ?

File: src/source/authentication.rst [hunk 4]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
...
+one that confirms hte user identity through the use of some kind of credentials (usually username/password).
```
Comment (twessling-icas) [inline src/source/authentication.rst]:
```suggestion
one that confirms the user identity through the use of some kind of credentials (usually username/password).
```

File: src/source/authentication.rst [hunk 5]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
   In this context *authorization server* is the server hosting authentication
```
Comment (twessling-icas) [inline src/source/authentication.rst]: shoud this be the 'authorization and token endpoints'? as authentication happens @ the tenant

File: src/source/authentication.rst [hunk 6]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
...
   and token endpoints.
 
+In our Oauth2 flow we rely on the *tenant* as being the **Authentication Server**, the
+one that confirms hte user identity through the use of some kind of credentials (usually username/password).
+The authentication server represents a user or a service account to the **Authorization Server**.
```
Comment (twessling-icas) [inline src/source/authentication.rst]: what does 'represents' mean in this context? I don't understand this sentence.

File: src/source/authentication.rst [hunk 7]
```diff
@@ -195,6 +202,8 @@ specified in the request at step 1 with the following GET parameters
     GET /code?code=AUTH_CODE&state=YOUR_STATE
     Host: yoursite.com
 
+This is how your Authorization server gets the short-lived code to further obtain a token.
```
Comment (twessling-icas) [inline src/source/authentication.rst]: your Authorization server? Your server, right?

Comment (twessling-icas) [reply src/source/authentication.rst]: Also, 'to subsequently get a refresh & access token'.

File: src/source/authentication.rst [hunk 8]
```diff
@@ -235,7 +244,15 @@ After obtaining the authorization code at step 2 the client needs to make a
     Host: admarkt.demo.qa-mp.so
     Content-Type: application/x-www-form-urlencoded
 
-    grant_type=authorization_code&code=AUTH_CODE&client_id=YOUR_CLIENT_ID&client_secret=YOUR_CLIENT_SECRET&redirect_uri=https://yoursite.com/code
+    -d 'grant_type=authorization_code&code=AUTH_CODE&client_id=YOUR_CLIENT_ID&client_secret=YOUR_CLIENT_SECRET&rredirect_uri=https://yoursite.com/code' https://admarkt.marktplaats.nl/accounts/oauth/token
+.. warning::
+    Beware of the Content-type: build it as a `multipart/form-data`.
```
Comment (twessling-icas) [inline src/source/authentication.rst]: the example above makes ut application/x-www-form-urlencoded, this is not in sync with this warning

File: src/source/authentication.rst [hunk 9]
```diff
@@ -235,7 +244,15 @@ After obtaining the authorization code at step 2 the client needs to make a
...
+
+    Beware also that you need to know the `user_id` on behalf of which the Authorization server accesses data. You can
+    obtain this through `passing the proper cookie: <https://pages.github.es.ecg.tools/ecg-icas/api/doc/tenant/api/post_auth_cookie.html>`_
+    from the user's browser (since it's on the same domain as the tenant).
+    Check the `tenant documentation: <https://pages.github.es.ecg.tools/ecg-icas/api/doc/tenant/api/get_config.html>`_ on how to get the `user_id` and which are the relevant cookies to configure for this.
```
Comment (twessling-icas) [inline src/source/authentication.rst]: hold up, this is external documentation referring to our internal documentation? This block shoudl not be here. 
The customer should never know the 'user_id'.

**Review by twessling-icas — CHANGES_REQUESTED** [review]

File: src/source/authentication.rst (hunk 3 above)
Comment (remenska) [reply src/source/authentication.rst]: how about *tenant* (local market) :D

Comment (remenska) [reply src/source/authentication.rst]: what's the difference? the local market plays a tenant role, that's how I see it (for us)

File: src/source/authentication.rst (hunk 6 above)
Comment (remenska) [reply src/source/authentication.rst]: presents?owns?

File: src/source/authentication.rst (hunk 8 above)
Comment (remenska) [reply src/source/authentication.rst]: indeed we're not doing curl examples

File: src/source/authentication.rst (hunk 9 above)
Comment (remenska) [reply src/source/authentication.rst]: then how do we get tenants to behave as sellside API partners?

Comment (remenska) [reply src/source/authentication.rst]: ah, nvm I see your point, will remove it

Comment (remenska) [reply src/source/authentication.rst]: >  You can obtain this through `passing the proper cookie:

this is not the API client actually. referred to as "You"

File: src/source/authentication.rst [hunk 10]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
   In this context *authorization server* is the server hosting authentication
   and token endpoints.
 
+In our Oauth2 flow we rely on the *tenant* (local market) as being the **Authentication Server**, the
+one that confirms the user identity through the use of some kind of credentials (usually username/password).
+The authentication server owns a user or a service account to the **Authorization Server**.
```
Comment (twessling-icas) [inline src/source/authentication.rst]: I don't fully understand the 'account to the auth server' part; how should I read this?

File: src/source/authentication.rst (hunk 5 above)
Comment (twessling-icas) [reply src/source/authentication.rst]: can you verify whether this is indeed correct or not?

**Review by twessling-icas — COMMENTED** [review]
minor questions remaining

File: src/source/index.rst [hunk 11]
```diff
@@ -48,6 +48,9 @@ History
 
 ..  role:: strike
 
+* v.0.123 - 09/09/2022 - dr
```
Comment (bakotaco) [inline src/source/index.rst]: consistent date formatting
```suggestion
* v.0.123 - 9 Sep 2022 - dr
```

File: src/source/authentication.rst [hunk 12]
```diff
@@ -40,6 +40,11 @@ OAuth 2.0 defines the following roles of users and applications:
   In this context *authorization server* is the server hosting authentication
   and token endpoints.
 
+In our Oauth2 flow we rely on the *tenant* (local market) as being the **Authentication Server**, the
```
Comment (bakotaco) [inline src/source/authentication.rst]: typo

```suggestion
In our OAuth 2.0 flow we rely on the *tenant* (local market) as being the **Authentication Server**, the
```

File: doc/next/search.html [hunk 13]
```diff
@@ -126,7 +130,7 @@
 <li class="toctree-l2"><a class="reference internal" href="feeds.html#character-encoding-line-terminators">Character encoding &amp; Line terminators</a></li>
 <li class="toctree-l2"><a class="reference internal" href="feeds.html#whitelisting">Whitelisting</a></li>
 <li class="toctree-l2"><a class="reference internal" href="feeds.html#example-xml">Example XML</a></li>
-<li class="toctree-l2"><a class="reference internal" href="feeds.html#scenario-s-questions">Scenario&#8217;s &amp; Questions</a></li>
+<li class="toctree-l2"><a class="reference internal" href="feeds.html#scenario-s-questions">Scenario’s &amp; Questions</a></li>
```
Comment (bakotaco) [inline doc/next/search.html:133]: do you know why these changes are triggered? (also some styling changes etc)

File: src/source/authentication.rst [hunk 14]
```diff
@@ -195,6 +202,8 @@ specified in the request at step 1 with the following GET parameters
     GET /code?code=AUTH_CODE&state=YOUR_STATE
     Host: yoursite.com
 
+This is how your server gets the short-lived code to subsequently get a refresh & access token.
```
Comment (bakotaco) [inline src/source/authentication.rst]: suggestion:

```suggestion
This is how your server gets the short-lived authorization code to subsequently get a refresh & access token.
```

File: src/source/authentication.rst (hunk 5 above)
Comment (remenska) [reply src/source/authentication.rst]: corrected

Comment (remenska) [reply src/source/authentication.rst]: good catch

File: src/source/authentication.rst (hunk 10 above)
Comment (remenska) [reply src/source/authentication.rst]: I meant a user or a service type of account. Removed, it just confuses :)

File: doc/next/search.html (hunk 13 above)
Comment (remenska) [reply doc/next/search.html:133]: eeh no clue

File: src/source/authentication.rst [hunk 15]
```diff
@@ -37,8 +37,12 @@ OAuth 2.0 defines the following roles of users and applications:
 
 * **Authorization Server:** The authorization server is the server
   authorizing the client app to access the resources of the resource owner.
-  In this context *authorization server* is the server hosting authentication
-  and token endpoints.
+  This allows fine-grained access to resources owned by the **resource owner**.
+
+In our OAuth 2.0 flow we rely on the *tenant* (local market) as being the **Authentication Server**, the
```
Comment (szpakas) [inline src/source/authentication.rst]: /s/as being/providing

File: src/source/authentication.rst [hunk 16]
```diff
@@ -37,8 +37,12 @@ OAuth 2.0 defines the following roles of users and applications:
...
+one that confirms the user identity through the use of some kind of credentials (usually username/password).
+The authentication server is hosting the authentication of a user account (resource owner).
```
Comment (szpakas) [inline src/source/authentication.rst:44]: minor/language: not sure you can say "hosting the authentication of". Maybe once more "providing" is better fitting?

File: src/source/authentication.rst [hunk 17]
```diff
@@ -195,6 +201,8 @@ specified in the request at step 1 with the following GET parameters
     GET /code?code=AUTH_CODE&state=YOUR_STATE
     Host: yoursite.com
 
+This is how your server gets the short-lived authorization code to subsequently get a refresh & access token.
```
Comment (szpakas) [inline src/source/authentication.rst]: It's not really "your server" but more "your client".

**Review by bakotaco — DISMISSED** [review]

File: src/source/authentication.rst (hunk 10 above)
Comment (remenska) [reply src/source/authentication.rst]: can we consider resolved? @twessling-icas

**Comment by twessling-icas** [comment]
The added image names our system 'iCAS' - I don't think our clients know us by this name? Should we make that Admarkt or whutnot?
Furthermore, I think adam has some nice minor suggestions still. Otherwise looks quite good imo

File: src/source/authentication.rst (hunk 17 above)
Comment (remenska) [reply src/source/authentication.rst]: spot on

**Comment by remenska** [comment]
> The added image names our system 'iCAS' - I don't think our clients know us by this name? Should we make that Admarkt or whutnot? Furthermore, I think adam has some nice minor suggestions still. Otherwise looks quite good imo

not sure about this really: is our product Admarkt even outside of BNL? happy to change it, domains are admarkt so it makes sense

File: doc/next/_sources/authentication.rst.txt [hunk 18]
```diff
@@ -407,3 +415,8 @@ granted by the user which is part of the user record.
  * - reporting
    - Grants access to reporting endpoints only. Allows creation of detailed downloadable reports
 
+
+You can find a generalized sequence diagram of the Oauth2 flow below (click to expand).
```
Comment (bakotaco) [inline doc/next/_sources/authentication.rst.txt]:
```suggestion
You can find a generalised sequence diagram of the OAuth 2.0 flow below (click to expand).
```

File: doc/next/_sources/index.rst.txt [hunk 19]
```diff
@@ -48,6 +48,9 @@ History
 
 ..  role:: strike
 
+* v.0.123 - 9 Sep 2022 - dr
+  * improved explanation on Oauth2 roles&flows, and added a sequence diagram
```
Comment (bakotaco) [inline doc/next/_sources/index.rst.txt]:
```suggestion
  * improved explanation on OAuth 2.0 roles & flows, and added a sequence diagram
```

File: src/source/authentication.rst [hunk 20]
```diff
@@ -407,3 +415,8 @@ granted by the user which is part of the user record.
  * - reporting
    - Grants access to reporting endpoints only. Allows creation of detailed downloadable reports
 
+
+You can find a generalized sequence diagram of the Oauth2 flow below (click to expand).
```
Comment (bakotaco) [inline src/source/authentication.rst]:
```suggestion
You can find a generalized sequence diagram of the OAuth 2.0 flow below (click to expand).
```

File: src/source/index.rst [hunk 21]
```diff
@@ -48,6 +48,9 @@ History
 
 ..  role:: strike
 
+* v.0.123 - 9 Sep 2022 - dr
+  * improved explanation on Oauth2 roles&flows, and added a sequence diagram
```
Comment (bakotaco) [inline src/source/index.rst]:
```suggestion
  * improved explanation on OAuth 2.0 roles&flows, and added a sequence diagram
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+
+ * - Scope
+   - ``api_ro`` or ``console_ro``
+
+ * - Accept
+   - ``application/sellside.ad-lite.list-v1+json``
+
+Returns an overview of ``adLite`` objects, based on matching query parameters, in the context of a particular campaign. In addition, it returns the total size of the
+result set matching the parameters criteria, and a next page token if additional pages are available.
+This view should be used to quickly get insight on the ads in a campaign that might need attention or optimizing, from an ad total budget perspective. It should not be used for scrapping the entire inventory of ads from that campaign.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:17]:
```suggestion
This view should be used to quickly get insight into the ads in a campaign that might need attention or optimizing, from an ad total budget perspective. It should not be used for scrapping the entire inventory of ads from that campaign.
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+At the moment ``adLite`` records are comprised of only the ad identifiers (by default sorted in descending order, newest ads first, unless specified otherwise), however, we plan to enhance the response with more denormalized data pertaining an ad, based on further understanding of how this endpoint will be used.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:18]:
```suggestion
At the moment ``adLite`` records are comprised of only the ad identifiers (by default sorted in descending order, newest ads first, unless specified otherwise), however, we plan to enhance the response with more denormalized data pertaining to an ad, based on further understanding of how this endpoint will be used.
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+
+Parameters
+~~~~~~~~~~
+
+=====================  ============     ================================================================================================================================================================================================================================
+Name                    Type             Description
+=====================  ============     ================================================================================================================================================================================================================================
+limit                  int               Limits the number of adLite records returned. Default **and** maximum is 100.
+categoryIds            list of ints      Comma-separated list of category id's to filter by. Only leaf category id's are useful, since ads can only be placed in leaf categories.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:27]:
```suggestion
categoryIds            list of ints      Comma-separated list of category IDs to filter by. Only leaf category IDs are useful since ads can only be placed in leaf categories.
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+regionIds              list of ints      Comma-separated list of region id's to filter by. RegionIds are not a concept used by all tenants, check before usage.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:28]:
```suggestion
regionIds              list of ints      Comma-separated list of region IDs to filter by. RegionIds are not a concept used by all tenants, check before usage.
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+pageToken              string            Skips the first N adLite records.
+remainingBudgetCents   int               Returns ads that have less-than, or equal, amount of cents of remaining total budget. By definition, it takes into account only ads that have set a total budget limit. Optional; no filtering applied if not provided.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:30]:
```suggestion
remainingBudgetCents   int               Returns ads that have less-than, or equal, amount of cents of the remaining total budget. By definition, it takes into account only ads that have set a total budget limit. Optional; no filtering applied if not provided.
```

File: doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+
+ * - Scope
+   - ``api_ro`` or ``console_ro``
+
+ * - Accept
+   - ``application/sellside.ad-lite.list-v1+json``
+
+Returns an overview of ``adLite`` objects, based on matching query parameters, in the context of a particular campaign. In addition, it returns the total size of the
+result set matching the parameters criteria.
+This view should be used to quickly get insight on the ads in a campaign that might need attention or optimizing, from an ad performance perspective.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt:17]:
```suggestion
This view should be used to quickly get insight into the ads in a campaign that might need attention or optimization, from an ad performance perspective.
```

File: doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt [hunk 28]
```diff
@@ -0,0 +1,68 @@
...
+
+The performance view always operates on a particular requested date range (by default: the past 7 days), and has no pagination possibilities at the moment.
+It should not be used for scrapping the entire inventory of ads from that campaign.
+At the moment ``adLite`` records are comprised of only the ad identifiers, however, we plan to enhance the response with more denormalized data pertaining an ad, based on further understanding of how this endpoint will be used.
```
Comment (bakotaco) [inline doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt:21]:
```suggestion
At the moment ``adLite`` records are comprised of only the ad identifiers, however, we plan to enhance the response with more denormalized data pertaining to an ad, based on further understanding of how this endpoint will be used.
```

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt (hunk 22 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:17]: this is completely unrelated to oauth2, and we'll likely revisit these APIs anyways

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt (hunk 25 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:28]: ditto

File: doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt (hunk 27 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt:17]: ditto

File: doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt (hunk 28 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-performance.rst.txt:21]: ditto

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt (hunk 23 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:18]: ditto

**Review by bakotaco — APPROVED** [review]

**Comment by twessling-icas** [comment]
> > The added image names our system 'iCAS' - I don't think our clients know us by this name? Should we make that Admarkt or whutnot? Furthermore, I think adam has some nice minor suggestions still. Otherwise looks quite good imo
> 
> not sure about this really: is our product Admarkt even outside of BNL? happy to change it, domains are admarkt so it makes sense

Good point, I'm honestly not sure what the best would be - and this repository is still ecg-icas/icas... For now we can probably keep it then, and let's put this forward as well to talk about our (tech) representation outward; we're going to onboard more german customers and other tenants will probably follow in the coming years... Best to have a single name for our platform. If that will be iCAS, all good, but let's make it a conscious decision and be consistent with it in our docs.

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt (hunk 22 above)
Comment (twessling-icas) [reply doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:17]: Also, scrapping -> scraping, this shows in a couple more places

**Comment by twessling-icas** [comment]
scrap the scrapping, use scraping instead :trollface:

**Comment by remenska** [comment]
> scrapping

not 100% sure what you mean :D but I'm not scraping anything related to campaign APIs right now :D

**Review by bakotaco — DISMISSED** [review]

**Comment by twessling-icas** [comment]
you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.`
I believe this should be `scraping` and not `scrapping`

**Comment by remenska** [comment]
> you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.` I believe this should be `scraping` and not `scrapping`

yes, it's on multiple places with those advanced views APIs (that will be revisited) and are out of scope of the goal of this PR

**Comment by twessling-icas** [comment]
the diagram still mentions POST /auth/cookie, I'd rather see this less transparent... like <validate login> or smth like that.
Next to that, the diagram uses channable.com as an example third-party; do we want to keep this or should we introduce our own at "your-company.com/code" etc?

**Comment by twessling-icas** [comment]
> > you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.` I believe this should be `scraping` and not `scrapping`
> 
> yes, it's on multiple places with those advanced views APIs (that will be revisited) and are out of scope of the goal of this PR

that is fine, but the typo remains.

**Comment by remenska** [comment]
> > > you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.` I believe this should be `scraping` and not `scrapping`
> > 
> > 
> > yes, it's on multiple places with those advanced views APIs (that will be revisited) and are out of scope of the goal of this PR
> 
> that is fine, but the typo remains.

yes, and I'm absolutely sure there are other typos in our documentation, but it is outside of scope (unless I missed the typo inside the oauth2 part). If you want, I can open a separate ticket (don't want to address those endpoints now for reasons I stated above)

**Comment by twessling-icas** [comment]
> > > > you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.` I believe this should be `scraping` and not `scrapping`
> > > 
> > > 
> > > yes, it's on multiple places with those advanced views APIs (that will be revisited) and are out of scope of the goal of this PR
> > 
> > 
> > that is fine, but the typo remains.
> 
> yes, and I'm absolutely sure there are other typos in our documentation, but it is outside of scope (unless I missed the typo inside the oauth2 part). If you want, I can open a separate ticket (don't want to address those endpoints now for reasons I stated above)

Ah, right, because with the later changes of this PR those sentences remain untouched. Fine, leave it for next time :)

**Comment by remenska** [comment]
> > > > > you've got some sentences like `It should not be used for scrapping the entire inventory of ads from that campaign.` I believe this should be `scraping` and not `scrapping`
> > > > 
> > > > 
> > > > yes, it's on multiple places with those advanced views APIs (that will be revisited) and are out of scope of the goal of this PR
> > > 
> > > 
> > > that is fine, but the typo remains.
> > 
> > 
> > yes, and I'm absolutely sure there are other typos in our documentation, but it is outside of scope (unless I missed the typo inside the oauth2 part). If you want, I can open a separate ticket (don't want to address those endpoints now for reasons I stated above)
> 
> Ah, right, because with the later changes of this PR those sentences remain untouched. Fine, leave it for next time :)

yes exactly, we're re-address those APIs anyways and I'll open a ticket for the typos :D I think what you stumbled on was autogenerated txt files or so

**Comment by remenska** [comment]
@twessling-icas can I get unblocked?

**Comment by twessling-icas** [comment]
> @twessling-icas can I get unblocked?

I also had some more comments on the png image, it mentions channable.com a lot, and still shows the POST /auth/cookie

**Comment by remenska** [comment]
> > @twessling-icas can I get unblocked?
> 
> I also had some more comments on the png image, it mentions channable.com a lot, and still shows the POST /auth/cookie

POST /auth/cookie i wanna show because it;s useful, as we get more and more tenants play the intermediary role of API partners. It doesn't hurt a lot

**Comment by remenska** [comment]
> > > @twessling-icas can I get unblocked?
> > 
> > 
> > I also had some more comments on the png image, it mentions channable.com a lot, and still shows the POST /auth/cookie
> 
> POST /auth/cookie i wanna show because it;s useful, as we get more and more tenants play the intermediary role of API partners. It doesn't hurt a lot

what's wrong with channable as example of a well known third party? But fine I'll make the final last change and I hope this PR doesnt linger and lose value further

**Comment by remenska** [comment]
done

**Review by bakotaco — APPROVED** [review]

**Review by twessling-icas — APPROVED** [review]

File: doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt (hunk 22 above)
Comment (remenska) [reply doc/0.123/_sources/reference/get-campaign-views-budget.rst.txt:17]: indeed, but all those generated txt/pdf files have nothing to do with my change :)
